UniqueArrayImp.h

```diff
@@ -0,0 +1,116 @@
+#ifndef MTMPARKINGLOT_UNIQUEARRAYIMP_H
+#define MTMPARKINGLOT_UNIQUEARRAYIMP_H
+
+template <class Element, class Compare>
+UniqueArray<Element, Compare>::UniqueArray(unsigned int size) :
+    data(new Element*[size]()), size(size) {}
+
+template <class Element, class Compare>
+UniqueArray<Element, Compare>::UniqueArray(const UniqueArray& other) :
+    data(new Element*[other.size]()), size(other.size)
+{
+    for (unsigned int i = 0; i < size; ++i) {
+        if (other.data[i]) {
```

Comment:
Does that check for a non NULL pointer?
I'm just not sure that a NULL is evaluated as a boolean False.

Reply:
It is, I even double-checked online. I think they even used this fact in some of their examples, but I'm not 100% sure.